MotionPhotoReader/app/src/main/java/com/google/android/libraries/motionphotoreader/HomographyMatrix.java

```diff
@@ -0,0 +1,142 @@
+package com.google.android.libraries.motionphotoreader;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static com.google.android.libraries.motionphotoreader.Constants.EPS;
+import static com.google.android.libraries.motionphotoreader.Constants.IDENTITY;
+
+/**
+ * Represents a 3x3 homography transformation in row-major matrix form.
+ *
+ * TODO: write unit test
+ */
+class HomographyMatrix {
+
+    private List<Float> matrix = new ArrayList<>();
+
+    public HomographyMatrix() {
+        for (float f : IDENTITY) {
+            matrix.add(f);
+        }
+    }
```

Comment:
```
matrix.addAll(IDENTITY)
```

Reply:
I thought this would work originally too, but it didn't compile. IDENTITY is a float array, is that why? I also tried converting IDENTITY to a collection but that ended up being longer than just adding each element individually.

Reply:
hmm.. how about `Arrays.asList( ... )` ?